### PR TITLE
Feature/yay

### PR DESCRIPTION
**Pull request by kewlfft**
yay helper added, yay does not support --aur for the moment but this is not preventing normal usage, the option has just no effect

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Currently, we support the following AUR helpers:
 
 - pacaur (recommended, default)
 - yaourt
+- yay
```
Comment (cdown) [inline README.md:7]: Please mention lack of auronly support

**Review by cdown — CHANGES_REQUESTED** [review]
yay with auronly=yes currently silently does the wrong thing. To avoid this, auronly=yes must be totally invalid with yay -- we should fail explicitly ourselves.

File: README.md (hunk 1 above)
Comment (cdown) [reply README.md:7]: (also, please consider filing a bugs with yay people about lack of support for AUR only upgrades, and also that they silently ignore wrong flags)

**Comment by kewlfft** [comment]
Yes indeed yay is not very mature yet on the command line . Agreed thanks for the feedback

**Comment by kewlfft** [comment]
please see my proposal, we explicitely ignore auronly with yay in both the doc and the code

File: aur.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def update_packages(module, tool, auronly):
     assert tool in TOOL_CMD_MAP
 
     cmd = ['env', 'LC_ALL=C'] + TOOL_CMD_MAP[tool] + ['-Su']
-    if auronly:
+    if auronly and (tool != 'yay'):
```
Comment (cdown) [inline aur.py]: This isn't what I meant -- we should fail hard (with an error) if auronly is set but we're using yay

**Review by cdown — CHANGES_REQUESTED** [review]

File: aur.py (hunk 2 above)
Comment (kewlfft) [reply aur.py]: This simply ignores it, this is what readme describes as well. If we should hard fail, to be consistent we should also hard fail whenever recurse is set to False and state present is used or any other improper or unsupported choice. What do you think? Is it what you suggest?

**Comment by kewlfft** [comment]
Not to pollute the functions, I now overwrite auronly for yay. This is consistent with ignoring it and not requiring to explicitely set auronly=no whenever using yay. Failing on the combination (auronly=yes and tool=yay) would require every call to yay to set auronly to no, heavy and unecessary for yay users.

**Comment by kewlfft** [comment]
yay has many command line issues, unless you think otherwise I will start with trizen's integration

**Pull request closed without merging**